packages/core/src/routes/session.ts

```diff
@@ -34,11 +35,34 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
             passwordEncrypted && passwordEncryptionMethod && passwordEncryptionSalt,
             'session.invalid_sign_in_method'
           );
-          assertThat(
-            encryptPassword(id, password, passwordEncryptionSalt, passwordEncryptionMethod) ===
-              passwordEncrypted,
-            'session.invalid_credentials'
-          );
+
+          if (
+            encryptPassword(id, password, passwordEncryptionSalt, passwordEncryptionMethod) !==
+            passwordEncrypted
+          ) {
+            ctx.userLogs = [
```

Comment:
also would like to check for the `ctx.userLogs` is it suppose to be immutable? What would be the downcost if we simpply call `ctx.userLogs.push()`

Reply:
one more thing cross my mind. maybe we can define a customized UserLog class.  and it can handle the default `payload` value and auto generate the `createdAt` at the constructor i.e.

either. ctx.userLogs.push(new UserLog({userId, type, result)):
or simply ctx.userLogs.add({userid, type, result});

Reply:
maybe because of the lint rule. if we have to call `.push` and the lint shows error, we edit the rule.

Reply:
Yes, is the lint rule.